wordpress-mobile/WordPress-Android
# UI Modernization Posts: Handle FAB action

Closes #19342 

This PR updates the FAB button on the Posts List to launch Create Post on tap
- Removes bottom sheet with multiple options
- Cleans up redundant code
- Update unit tests
- Add track event for FAB click

**To test:**
- Install the app
- Login and select a site
- Navigate to My Site > Posts
- Tap the FAB button
- ✅ Verify the bottom sheet is no longer shown and you are taken directly to the Create Post view
- ✅ Verify logs contain  🔵 Tracked: post_list_create_post_tapped, Properties: {"action":"create_new_post"}

## Regression Notes
1. Potential unintended areas of impact
The FAB does not work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit tests

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:

UI Changes testing checklist: N/A
